src/fparser/readfortran.py

```diff
@@ -1048,47 +1034,47 @@ def get_source_item(self):
                               ' of fixed Fortran code' % (line[i],i+1)
                     if i==0:
                         message += ', interpreting line as comment line'
-                    if self.isfix:     
+                    if self.isfix:
                         if i!=0:
                             message = message + ', switching to free format mode'
                         message = self.format_warning_message(\
                             message,startlineno, self.linecount)
-                        logger.warning(message)
-                        # self.show_message(message, sys.stderr)
+                        logging.getLogger(__name__).warning(message)
                         if i==0:
                             # non standard comment line:
-                            return self.comment_item(line, startlineno, startlineno)                           
+                            return self.comment_item(line, startlineno, startlineno)
                         self.set_mode(True, False)
                     else:
+                        message = self.format_warning_message(message,
```

Comment:
The logic is a little convoluted, it took me a while to work out what was going on. I'm still not sure I fully understand it.

If we are dealing with fixed format source, the first 6 characters are checked for the special things they might contain. The line has already been checked for comments and a return made so we don't need to worry about that.

If any of those first 6 characters are not a space or a digit we have found something we shouldn't. A message is prepared explaining this but it is not logged yet. When it is the first character the message is extended to explain that we assume it's a comment.

If we are in "fix" mode the message us further extended to note that we are switching to free format mode. At this point the message is logged.

However, if we are not in "fix" mode then it falls through the section we are discussing here. Originally the message would only be logged if it was the first character, but it is valid for any occurrence of an illegal character in the first 6 places. Regardless of whether the message was logged or not the routine would return with something although I'm not entirely sure it makes sense to do so.

This is a long way of saying that the reason I moved the logging stage is so that the message would be logged regardless of whether it was the first character or not which was in error.